pkg/workload/ycsb/ycsb.go

```diff
@@ -937,6 +963,10 @@ func (yw *ycsbWorker) chooseOp() operation {
 		return insertOp
 	}
 	p -= yw.config.insertFreq
+	if readOnly.Load() == 0 && p <= yw.config.deleteFreq {
+		return deleteOp
+	}
+	p -= yw.config.deleteFreq
```

Comment:
I don't get what this subtraction logic is doing. It seems to be ruining uniform distribution :) E.g., say `p = 0.25` (drawn from uniform distribution), `deleteFreq = 0.20` and `readModifyWriteFreq = 0.05`, which means we end up with `readModifyWriteOp` when we really shouldn't.

Reply:
Why shouldn't we end up with `readModifyWriteOp` in that case? We assume in this method that the frequencies are normalized so that they sum to 1.0. We then pick a random float from `[0.0,1.0)`. The subtraction logic is implementing the weighted mapping from random float to random operation.

Reply:
Yeah, it works. I didn't immediately see that the mapping is from ($`s - \sum p_j`$) to $`p_{j+1}`$, where $`s`$ is sampled from $`[0, 1)`$.

Reply:
Added a more robust comment.  Hopefully this resolves the confusion.